flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaInternalProducerITCase.java

```diff
@@ -204,7 +204,7 @@ public void testFlushAfterClosed() {
 
     @Test(timeout = 30000L)
     public void testProducerWhenCommitEmptyPartitionsToOutdatedTxnCoordinator() throws Exception {
-        String topic = "flink-kafka-producer-txn-coordinator-changed";
+        String topic = "flink-kafka-producer-txn-coordinator-changed-" + UUID.randomUUID();
```

Comment:
Just an example code location where I don't understand why we have to add randomness. As far as I understand now (and based on your previous comment), we're adding the randomness in cases where a test retry is potentially triggered due to some flakiness. I would understand if a topic is reused in other tests which is also not the case here. 
It feels that I'm missing something here because I don't see none of the reasonings I could come up with being applied here, for instance. There are a few other tests where I'm struggling to find the reason as well.

Reply:
I think the point here is that adding randomness in general doesn't hurt, it might not be strictly necessary everwhere but it's easier to add it everywhere to fix the test instability which is a very urgent matter.

Let's go ahead with the current implementation, if someone want's to improve this further we can do it once the instabilities are gone.